### PR TITLE
boot gen4 spider from UFS

### PR DESCRIPTION
**Pull request by sa-kib**
Migrate the complete system for boot from UFS and leave MMC for dom0 internal purposes only. This requires bl33 to support UFS, domd UFS support and generating a proper image for UFS.

### CODE REVIEW

**Comment by sa-kib** [comment]
companion dom0 PR: https://github.com/aoscloud/aos_core_zephyr/pull/65

File: README.md [hunk 1]
```diff
@@ -73,5 +73,8 @@ R-Car Gen3 device.
 To deploy the full image to R-Car-S4 device, see [meta-aos-rcar-gen4]
 (https://github.com/aoscloud/meta-aos-rcar-gen4/blob/main/README.md).
 
+Additionaly for UFS on R-Car-S4 following command set_ufs has to be executed before system boot:
+setenv set_ufs 'i2c dev 0; i2c mw 0x6c 0x26 0x05 ;i2c olen 0x6c 2; i2c mw 0x6c 0x13a 0x86 ;i2c mw 0x6c 0x268 0x06; i2c mw 0x6c 0x269 0x00; i2c mw 0x6c 0x26a 0x3c; i2c mw 0x6c 0x26b 0x00; i2c mw 0x6c 0x26c 0x06; i2c mw 0x6c 0x26d 0x00; i2c mw 0x6c 0x26e 0x3f; i2c mw 0x6c 0x26f 0x00'
+
 To deploy the full image to R-Car Gen3 device, see [meta-aos-rcar-gen3]
-(https://github.com/aoscloud/meta-aos-rcar-gen3/blob/main/README.md).
\ No newline at end of file
+(https://github.com/aoscloud/meta-aos-rcar-gen3/blob/main/README.md).
```
Comment (oleksiimoisieiev) [inline README.md:80]: Looks like unnecessary change. Please remove.

File: README.md [hunk 2]
```diff
@@ -73,5 +73,8 @@ R-Car Gen3 device.
 To deploy the full image to R-Car-S4 device, see [meta-aos-rcar-gen4]
 (https://github.com/aoscloud/meta-aos-rcar-gen4/blob/main/README.md).
 
+Additionaly for UFS on R-Car-S4 following command set_ufs has to be executed before system boot:
+setenv set_ufs 'i2c dev 0; i2c mw 0x6c 0x26 0x05 ;i2c olen 0x6c 2; i2c mw 0x6c 0x13a 0x86 ;i2c mw 0x6c 0x268 0x06; i2c mw 0x6c 0x269 0x00; i2c mw 0x6c 0x26a 0x3c; i2c mw 0x6c 0x26b 0x00; i2c mw 0x6c 0x26c 0x06; i2c mw 0x6c 0x26d 0x00; i2c mw 0x6c 0x26e 0x3f; i2c mw 0x6c 0x26f 0x00'
```
Comment (oleksiimoisieiev) [inline README.md:77]: I think aos bootargs should be updated as well. Please contact @al1img to get aos bootargs updated

**Comment by oleksiimoisieiev** [comment]
Also please follow commit message format, which is used for this repo

File: README.md (hunk 2 above)
Comment (al1img) [reply README.md:77]: Better to put it in seperate chapter:
```
## Enable UFS
```

Why so many magic digits? I guess, one small mistake and UFS stops working. Is there any other  solution to enable it?

File: meta-aos-rcar-gen4/meta-aos-rcar-gen4-domd/recipes-bsp/u-boot/files/0001-arm-dts-r8a779f0-Add-Renesas-UFS-HCD-support.patch [hunk 3]
```diff
@@ -0,0 +1,78 @@
+From decc9593a8413e9eb3b0133c11072aed2a61a418 Mon Sep 17 00:00:00 2001
```
Comment (al1img) [inline meta-aos-rcar-gen4/meta-aos-rcar-gen4-domd/recipes-bsp/u-boot/files/0001-arm-dts-r8a779f0-Add-Renesas-UFS-HCD-support.patch:1]: Where this patch comes from? Why it is in patch form?

File: README.md (hunk 2 above)
Comment (sa-kib) [reply README.md:77]: this is done according to vendor's guide. Hopefully it will change in later releases, and either UFS will auto-powerup on boot, or bootloader will do it. This script is temporary.

Also this guide is specific to S4 only, hence it's put into gen4 deployment chapter. Not related to generic UFS deployments though.

File: meta-aos-rcar-gen4/meta-aos-rcar-gen4-domd/recipes-bsp/u-boot/files/0001-arm-dts-r8a779f0-Add-Renesas-UFS-HCD-support.patch (hunk 3 above)
Comment (sa-kib) [reply meta-aos-rcar-gen4/meta-aos-rcar-gen4-domd/recipes-bsp/u-boot/files/0001-arm-dts-r8a779f0-Add-Renesas-UFS-HCD-support.patch:1]: the patch is from me, comes as a separate patch because it's not present in vendor's upstream repo yet

File: README.md (hunk 1 above)
Comment (sa-kib) [reply README.md:80]: git won't let me remove it :'(

**Comment by sa-kib** [comment]
> Also please follow commit message format, which is used for this repo

done

**Comment by oleksiimoisieiev** [comment]
+1 Acked-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>

File: README.md [hunk 4]
```diff
@@ -73,5 +73,10 @@ R-Car Gen3 device.
 To deploy the full image to R-Car-S4 device, see [meta-aos-rcar-gen4]
 (https://github.com/aoscloud/meta-aos-rcar-gen4/blob/main/README.md).
 
+Additionaly for UFS on R-Car-S4 following command set_ufs has to be executed before system boot:
+
+```sh
```
Comment (al1img) [inline README.md:78]: You forgot to close code block.

Comment (sa-kib) [reply README.md:78]: indeed, fixed

**Review by al1img — APPROVED** [review]
